src/nvim/indent_c.c

```diff
@@ -508,29 +505,25 @@ get_indent_nolabel (     /* XXX */
  *   label:	if (asdf && asdfasdf)
  *		^
  */
-static int skip_label(linenr_T lnum, char_u **pp)
-{
-  char_u      *l;
-  int amount;
-  pos_T cursor_save;
+static int get_indent_after_label_pp(linenr_T lnum, char_u **pp)
+  FUNC_ATTR_NONNULL_ALL
+{ /* XXX */
+  char_u *l = ml_get(lnum);
+  int indent;
 
-  cursor_save = curwin->w_cursor;
-  curwin->w_cursor.lnum = lnum;
-  l = get_cursor_line_ptr();
-  /* XXX */
+  // We repeatedly call "ml_get(lnum)" because every time it's called (or any
+  // time we call a function that calls it, marked with an 'XXX' comment) the
+  // returned pointer is invalidated.
```

Comment:
We started to change the documentation style for functions (doxygen). New functions always follow this style and old ones could be updated. Have a look at memory.c for examples.  

> I didn't find any tabs here to change to spaces, can you tell me which tabs you mean?

The function description has an example, which contains tabs:
```c
/*
 * Find indent for line "lnum", ignoring any case or jump label.
 * Also return a pointer to the text (after the label) in "pp".
 *   label:	if (asdf && asdfasdf)
 *		^
 */
```
The arrow (`^`) is not aligned with the `if` for me, because tabs are used for indentation and their size depends on option `'tabstop'`.


> Weirdly, the convention throughout the file is to add the /* XXX */ comment whenever you might invalid an ml_get_*() pointer. It took me forever to realize this though.

In my opinion, with a note or warning in the function description about the invalidation of pointers, the meaning of `/* XXX */` is easier to get and better documented. But I would leave the `/* XXX */` in the code until all affected function would have been updated.